src/plugins/new-ui/public/webcomponents/compass.html

```diff
@@ -48,143 +84,265 @@
           </g>
         </g>
       </svg>
     </div>
+
+    
   </template>
+  
   <script>
-    var lastData = null;
-    var newData = null;
-    Polymer({
-      is: 'orov-newui-compass',
-      properties: {
-        bearing: {
-          observer: 'setBearing',
-          notify: true
+      var lastData = null;
+      var newData = null;
+
+      //The operating mode of the display
+      var isCompass = true;
+
+      var mapTo = function (value, in_min, in_max, out_min, out_max) {
+        return (value - in_min) * (out_max - out_min) / (in_max - in_min) + out_min;
+      };
+
+      //Work around because javascript does not use pass by reference for primitaves
+      //TODO: Is there a better, more idiomatic way?
+      var bearingAngle = { angle: null };
+      var indicatorAngle = { angle: null };
+
+      function rotateElement(element, obj, newRotation) {
+        var aR;
+
+        obj.angle = obj.angle || 0; // if rot undefined or 0, make 0, else rot
+
+        aR = obj.angle % 360;
+        if( aR < 0 )
+        {
+          aR += 360; 
+        }
+        if( aR < 180 && (newRotation > (aR + 180)) ) 
+        { 
+          obj.angle -= 360; 
+        }
+        if ( aR >= 180 && (newRotation <= (aR - 180)) ) 
+        { 
+          obj.angle += 360; 
+        }
+        obj.angle += (newRotation - aR);
+
+        element.style.transform = ("rotate( " + obj.angle + "deg )");
+      }
+
+      Polymer({
+        is: 'orov-newui-compass',
+        properties: {
+          bearing: {
+            type: String,
+            value: '000'
+          },
+          heading: {
+            type: String,
+            value: '000'
+
+            // value: function(){return '000';},
+            // observer: 'setHeading',
+            // notify: true
+          },
+          mode: {
+            type: String,
+            value: "Compass"
+          },
+          headingLock: {
+            type: Boolean,
+            value: false
+          },
+          options: {
+            type: Object,
+            value: {
+                indicator: 'white',
+                bezel: 'white',
+                bearing: 'white',
+                background: 'black',
+                headingFill: 'white',
+                modeFill: 'white',
+                // will be set to options.bezel
+                headingStrokeWidth: 0,
+                headingStrokeWidthTargetHold: 15
+              }
+          },
+          target: {
+            type: Number,
+            value: function(){return 0;}
+          }
+        },
+        listeners: {
+          'mode.tap': 'toggleMode',
+          'heading.tap': 'toggleHeadingLock'
+        },
+        toggleMode: function(e) {
+
+          //This is a toggle
+          if(isCompass)
+          {
+            this.$.mode.textContent = "Gyro";
```

Comment:
However, on reading further you are using a state isCompass to figure out what is happening. So you should simply create a helper method _compassMode() for instance, and use that in the template {{__(_compassMode())}} where the helper checks isCompass and return the right text.